File: lib/dart/birthday_cake_candles.dart
import 'dart:math' as math;

int birthdayCakeCandles(final List<int> candles) {
  final max = candles.reduce(math.max);
  return candles.where((final candle) => candle == max).length;
}

int birthdayCakeCandles1(final List<int> candles) {
  var maxValue = 0;
  final numbsCountMap = <int, int>{};

  for (final number in candles) {
    if (numbsCountMap.containsKey(number)) {
      numbsCountMap[number] = numbsCountMap[number]! + 1;
    } else {
      numbsCountMap[number] = 1;
    }
    maxValue = math.max(maxValue, numbsCountMap[number]!);
  }
  return maxValue;
}

File: lib/swift/birthday_cake_candles.swift
func birthdayCakeCandles1(candles: [Int]) -> {
    var maxValue = 0
    var numbsCountMap:[Int:Int] = [:]
    
    for number in candles {
        let keyExists = numbsCountMap[number]
        
        if let _ = keyExists {
            numbsCountMap[number]! += 1;
        } else {
            numbsCountMap[number] = 1;
        }
        maxValue = max(maxValue, number);
    }

    return numbsCountMap[maxValue]!;
}

